tests/php/Core/EnvironmentTest.php

```diff
@@ -66,4 +67,115 @@ public function testGetVariables()
         $this->assertEquals('fail', $vars['test']);
         $this->assertEquals('global', $GLOBALS['test']);
     }
+
+    public function provideHasEnv()
+    {
+        return [
+            [
+                'setAs' => '.env',
+                'value' => true,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '.env',
+                'value' => false,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '.env',
+                'value' => null,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_ENV',
+                'value' => true,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_ENV',
+                'value' => false,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_ENV',
+                'value' => null,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_SERVER',
+                'value' => true,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_SERVER',
+                'value' => false,
+                'expected' => true,
+            ],
+            [
+                'setAs' => '_SERVER',
+                'value' => null,
+                'expected' => true,
+            ],
+            [
+                'setAs' => 'putenv',
+                'value' => true,
+                'expected' => true,
+            ],
+            [
+                'setAs' => 'putenv',
+                'value' => false,
+                'expected' => true,
+            ],
+            [
+                'setAs' => 'putenv',
+                'value' => null,
+                'expected' => true,
+            ],
+            [
+                'setAs' => null,
+                'value' => null,
+                'expected' => false,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideHasEnv
+     */
+    public function testHasEnv(?string $setAs, ?bool $value, bool $expected)
```

Comment:
Should this test also test other types of values like text and numbers, not just boolean? I know `false` value was the main driver but if we're testing for a set env variable I'd say we should test with other types as well, despite them being less problematic than boolean false.

Reply:
It certainly can do. I thought about that at the time but decided all values that aren't boolean/null will effectively be treated truthy/falsy anyway. But I have no issues added in more. Beyond strings and ints do you have any other types you want me to add?

Reply:
Can't think of anything else at the moment. Anything more complex like objects etc we usually store serialised as json text or similar, so it's a string anyway and who and how uses it is not relevant here, so I think `0`, `1`, `text string` would do. How we treat an empty string? As not set? Or null?

Reply:
Empty string is set - it's explicitly set as an empty string.

Reply:
Yup, it's set. With the `putenv` example: `putenv('_ENVTEST_HAS_ENV=');` _is_ explicitly setting the value, as opposed to `putenv('_ENVTEST_HAS_ENV')` which clears it.